### ISSUE TITLE
Missing dependency on VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT

### ISSUE THREAD

**Issue by darksylinc** [LunarG/VulkanSamples#267]
According to [Synchronization Examples](https://github.com/KhronosGroup/Vulkan-Docs/wiki/Synchronization-Examples#swapchain-image-acquire-and-present), [Yet another blog explaining Vulkan synchronization](http://themaister.net/blog/2019/08/14/yet-another-blog-explaining-vulkan-synchronization/) and personal experience, using a semaphore is not enough.

A pipeline barrier or an input dependency must be used to guard the backbuffer e.g.

```
imageBarrier.srcAccessMask = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
imageBarrier.dstAccessMask =
    VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT | VK_ACCESS_COLOR_ATTACHMENT_READ_BIT;
imageBarrier.newLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
imageBarrier.oldLayout = VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
imageBarrier.image = textureVulkan->getFinalTextureName();

imageBarrier.subresourceRange = textureVulkan->getFullSubresourceRange();

vkCmdPipelineBarrier(
    mQueue->mCurrentCmdBuffer, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT,
    VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT, 0u, 0u, 0, 0u, 0, 1u, &imageBarrier );
```

or alternatively as a subpass dependency:

```
// Create a dependency to external events.
// We need to wait for the WSI semaphore to signal.
// Only pipeline stages which depend on COLOR_ATTACHMENT_OUTPUT_BIT will
// actually wait for the semaphore, so we must also wait for that pipeline stage.
VkSubpassDependency dependency = {0};
dependency.srcSubpass          = VK_SUBPASS_EXTERNAL;
dependency.dstSubpass          = 0;
dependency.srcStageMask        = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
dependency.dstStageMask        = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;

// Since we changed the image layout, we need to make the memory visible to
// color attachment to modify.
dependency.srcAccessMask = 0;
dependency.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
```

**Update:** Sorry for the brevety, I had to rush out while I typed this.
Basically all the samples never insert this depedency, and therefore are liable to bugs. I suspect this will never show up as a bug because all samples are single buffered, therefore the GPU will almost never end up accessing an image while it is still being presented.

**Comment by Tony-LunarG** [comment]
Yes, I think you're right. Note that most of these samples were written well before the synchronization chapter of the spec was written and they are subject to falling behind an evolving spec.  Thanks for finding this, and we will address it as time allows.

**Comment by Tony-LunarG** [comment]
Fixed with #270
